server/src/main/java/org/opensearch/indices/IndicesService.java

```diff
@@ -314,6 +314,17 @@ public class IndicesService extends AbstractLifecycleComponent
         Property.Final
     );
 
+    /**
+     * This setting is used to enable the optimisation in prefix path which helps in achieving higher throughput and lesser
+     * rate limiting by remote store providers. This setting is effective only for remote store enabled cluster.
+     */
+    public static final Setting<Boolean> CLUSTER_REMOTE_STORE_PATH_PREFIX_OPTIMISED_SETTING = Setting.boolSetting(
+        "cluster.remote_store.index.path.prefix.optimised",
+        false,
+        Property.NodeScope,
+        Property.Dynamic
```

Comment:
should this be Dynamic? do we anticipate it to be used in this way?

Reply:
This is good to have setting which will allow user to have both kind of blob store path possible. Once the cluster setting changes, the new indexes of the cluster will be created with the updated setting.

Reply:
`optimized` is the result of using  `HASHED_PREFIX` . Should we rename it to `index.path.prefix.hashed` ?

Reply:
As discussed, let me get one setting `index.path.prefix.type` which will have values like fixed, hashed_prefix. This will allow extensibility in case in future we decide to have more approaches for blob store path type.